### PR TITLE
ARROW-15680: [C++] Temporal floor/ceil/round should accept week_starts_monday when rounding to multiple of week

### PR DESCRIPTION
**Pull request by rok**
This is to resolve [ARROW-15680](https://issues.apache.org/jira/browse/ARROW-15680).

### CODE REVIEW

File: cpp/src/arrow/compute/kernels/scalar_temporal_unary.cc [hunk 1]
```diff
@@ -725,6 +725,27 @@ const Duration FloorTimePoint(const int64_t arg, const int64_t multiple,
   }
 }
 
+template <typename Duration, typename Unit, typename Localizer>
```
Comment (lidavidm) [inline cpp/src/arrow/compute/kernels/scalar_temporal_unary.cc:728]: nit: is there any point templating on Unit when it's apparently always `weeks` (and doesn't seem to make sense for any other unit)?

**Review by lidavidm — APPROVED** [review]
LGTM.

File: cpp/src/arrow/compute/kernels/scalar_temporal_unary.cc (hunk 1 above)
Comment (rok) [reply cpp/src/arrow/compute/kernels/scalar_temporal_unary.cc]: Done.

**Review by lidavidm — APPROVED** [review]

**Comment by lidavidm** [comment]
@rok I think it's https://github.com/conda-forge/abseil-cpp-feedstock/pull/25#issuecomment-1054403044

**Comment by lidavidm** [comment]
I think we can ignore these CI failures for now, merging.

**Pull request closed without merging**